opentasks-provider/src/main/java/org/dmfs/provider/tasks/handler/RelationHandler.java

```diff
@@ -44,21 +43,13 @@ public ContentValues validateValues(SQLiteDatabase db, long taskId, long propert
 
         Long id = values.getAsLong(Relation.RELATED_ID);
         String uid = values.getAsString(Relation.RELATED_UID);
-        String uri = values.getAsString(Relation.RELATED_URI);
 
-        if (id == null && uri == null && uid != null)
+        if (id == null && uid != null)
```

Comment:
Shouldn't this `validate()` method just simply throw if none or both of `RELATED_ID` and `RELATED_UID` are present?
Isn't the value already `null` when `values.putNull(Relation.RELATED_ID);` is called?

Reply:
that would only be the case for new rows. When you override a row we need to reset the other value explicitly. That being said, I already considered making rows write-once. So you can only create and delete them. Updates would throw an exception. I'm starting to get used to this idea and it would certainly make a couple of things much simpler.

Reply:
I still don't get this - so just by looking at this code:
```java
Long id = values.getAsLong(Relation.RELATED_ID);
String uid = values.getAsString(Relation.RELATED_UID);

if (id == null && uid != null)
{
    values.putNull(Relation.RELATED_ID);
}
```
and just ignore the 'uid' for a second and inline the 'id' variable, it becomes this:
```java
if (values.getAsLong(Relation.RELATED_ID) == null)
{
    values.putNull(Relation.RELATED_ID);
}
```
So if the value is `null` we set it to `null`..   Or do I miss something?

Reply:
`getAsLong` also returns `null` if the value is absent (i.e. not not present in the `ContentValues`), so we need to explicitly put a `null` value to ensure we override the old value, otherwise it would remain unchanged.

Reply:
oh, I see, thanks